### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by ildyria**
Lots of rewriting... Started writing comments.

### CODE REVIEW

File: app/Configs.php [hunk 1]
```diff
@@ -8,95 +8,80 @@
...
-	 * @param string $key
-	 * @param bool $public
-	 * @return bool
-	 */
-	static protected function inExceptArray(string $key, bool $public)
+	private static $cache = null;
+
+
+
+	public static function arrayify($query)
```
Comment (hermzz) [inline app/Configs.php]: Eloquent does something for this: https://laravel.com/docs/5.8/collections#method-pluck

Comment (ildyria) [reply app/Configs.php]: Niiiiice :astonished:  Thanks !

File: app/Http/Controllers/IndexController.php [hunk 2]
```diff
@@ -28,16 +28,17 @@ public function __construct(ConfigFunctions $configFunctions)
...
 		{
-			$lang = Lang::get_lang(Configs::where('key', '=', 'lang')->first());
+			$lang = Lang::get_lang(Configs::get_value('lang', 'en'));
 
 			$infos = $this->configFunctions->get_pages_infos();
 
 			$menus = Page::menu()->get();
 
-			$title = Configs::get_value('site_title');
+			$title = Configs::get_value('site_title', 'Lychee v4');
```
Comment (hermzz) [inline app/Http/Controllers/IndexController.php]: Given that `Lychee v4` is repeated a bunch of times, wouldn't it make sense to define it in one of the `config/` files?

Comment (ildyria) [reply app/Http/Controllers/IndexController.php]: good point.

File: app/Http/Controllers/PageController.php [hunk 3]
```diff
@@ -28,15 +28,17 @@ public function __construct(ConfigFunctions $configFunctions)
 
 	function page(Request $request, $page)
 	{
+		Configs::get();
```
Comment (hermzz) [inline app/Http/Controllers/PageController.php]: This is a bit awkward, is there a way to make `Configs::get_value` check if the cache is not set and call `Configs::get` from within?

Comment (ildyria) [reply app/Http/Controllers/PageController.php]: it is supposed to be the case but for some reason it is not working:
https://github.com/LycheeOrg/Lychee-Laravel/blob/refactoring/app/Configs.php#L66

File: app/ModelFunctions/SessionFunctions.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+	 */
+	public function noLogin()
+	{
+
+		$configs = Configs::get();
+
+		// Check if login credentials exist and login if they don't
+		if (isset($configs['username']) && $configs['username'] === '' &&
+			isset($configs['password']) && $configs['password'] === '') {
+			Session::put('login', true);
```
Comment (d7415) [inline app/ModelFunctions/SessionFunctions.php]: I don't like the indentation here, but I can't think of a better way either.

File: database/migrations/2019_03_08_135817_private_config_fields.php [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	 * @return void
+	 */
+	public function up()
+	{
+		if (Schema::hasTable('configs')) {
+			Schema::table('configs', function ($table) {
+				$table->tinyInteger('confidentiality')->after('cat')->default(0);
+			});
+
+			// set confidentiality to 3 for those (never returned)
```
Comment (d7415) [inline database/migrations/2019_03_08_135817_private_config_fields.php]: s/3/4/

**Pull request merged**